### PR TITLE
Upgrade dskit and pass empty service name to dskit/grpcutil.Resolver.Resolve

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does**:
Upgrade to dskit@01ce9286d7d5, which most importantly takes a new argument, `service`, to `dskit/grpcutil.Resolver.Resolve`. This argument configures the service part of the SRV DNS record to try and resolve. The PR passes an empty string for the aforementioned argument, so SRV lookup is disabled.

**Which issue(s) this PR fixes**:

**Checklist**
- [na] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by aknuds1** [comment]
I think the integration tests might be running with an older version of Go, since `os.ReadFile` isn't available?

**Comment by pstibrany** [comment]
Can we please update dskit to latest main, there are more incompatible changes (removal of ring keys from dskit), it would be best to tackle them all at once. WDYT?

**Comment by aknuds1** [comment]
> Can we please update dskit to latest main, there are more incompatible changes (removal of ring keys from dskit), it would be best to tackle them all at once. WDYT?

@pstibrany is it best to tackle that also in this PR? I honestly am not sure how to tackle those changes. If it's easily done I don't mind.

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,7 +76,7 @@
 * [BUGFIX] Querier: honor querier minT,maxT if `nil` SelectHints are passed to Select(). #4413
 * [BUGFIX] Compactor: fixed panic while collecting Prometheus metrics. #4483
 * [BUGFIX] Update go-kit package to fix spurious log messages #4544
-
+* [BUGFIX] Querier: Disable query scheduler SRV DNS lookup. #4601
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply CHANGELOG.md:79]: @bboreham good point, I can try to improve the changelog entry. I'm not so familiar with the other changes in dskit, I guess I ought to try to summarize them also?

**Comment by pstibrany** [comment]
> > Can we please update dskit to latest main, there are more incompatible changes (removal of ring keys from dskit), it would be best to tackle them all at once. WDYT?
> 
> @pstibrany is it best to tackle that also in this PR? I honestly am not sure how to tackle those changes. If it's easily done I don't mind.

It's relatively easy to do, we just need to stop using constants from dskit, and define them in Cortex. I will give you a pointer via Slack.

**Comment by aknuds1** [comment]
Thanks @pstibrany, coming up.

**Comment by aknuds1** [comment]
@pstibrany upgraded to latest dskit as you asked. @bboreham wrote more on the changelog entry, plus added the ones @pstibrany thought were useful from the dskit changelog.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aknuds1) [reply CHANGELOG.md]: I've expanded upon this changelog entry, and added ones from dskit @pstibrany thought were useful.

**Review by pstibrany — APPROVED** [review]
Good job, thank you!

**Review by 56quarters — APPROVED** [review]
LGTM

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**